### PR TITLE
Add SQFC support

### PR DESCRIPTION
**Pull request by PabstMirror**
Adds sqfc.json which controls which files gets compiled and what is required, for now using setup p:

<details>
  <summary>old fixed bugs</summary>

[ACE] (common) ERROR: Setting [ace_zeus_zeusAscension] - value type [BOOL] is unknown - []
[ACE] (common) ERROR: Setting [ace_advanced_ballistics_bulletTraceEnabled] - value type [BOOL] is unknown - []

Error in expression < "barStatement"));
        (_finalArray select 4) set [0, _barStatement];
    };>
  Error position: <select 4) set [0, _barStatement];
    };>
  Error Zero divisor
File /z/ace/addons/arsenal/functions/fnc_compileStats.sqf..., line 110
Error in expression <de = _long;
};

[_GZD, _grid100km, _GZD + _grid100km]
>
  Error position: <+ _grid100km]
>
  Error Generic error in expression
File /z/ace/addons/common/functions/fnc_getMGRSdata.sqf..., line 153

 [ACE] (interact_menu) ERROR: Failed to add action - action (ace_dragging_drag) to parent ["ACE_MainActions"] on object B_HeavyGunner_F [0]
 [ACE] (interact_menu) ERROR: Failed to add action - action (ace_dragging_carry) to parent ["ACE_MainActions"] on object B_HeavyGunner_F [0]
 Error in expression <_ActNamespace;

if (!isNil {_namespace getVariable _objectType}) exitWith {};

>
   Error position: <getVariable _objectType}) exitWith {};

>
   Error getvariable: Type Object, expected Array,String
 File /z/ace/addons/interact_menu/functions/fnc_compileMenu.sqf..., line 27
  âž¥ Context: 	[] L1 ()
	[] L1 ()
	[] L30 (x\cba\addons\xeh\fnc_init.sqf)
	[] L36 (x\cba\addons\xeh\fnc_init.sqf)
	[] L35 (x\cba\addons\xeh\fnc_init.sqf)
	[] L13 (z\ace\addons\interact_menu\XEH_preInit.sqf)
	[] L27 (/z/ace/addons/interact_menu/functions/fnc_compileMenu.sqf)

 Error in expression <ManActions"}) exitWith {
    _namespace setVariable [_objectType, +ace_interact_>
   Error position: <setVariable [_objectType, +ace_interact_>
   Error Type Object, expected String
 File /z/ace/addons/interact_menu/functions/fnc_compileMenu.sqf..., line 35
  âž¥ Context: 	[] L1 ()
	[] L1 ()
	[] L30 (x\cba\addons\xeh\fnc_init.sqf)
	[] L36 (x\cba\addons\xeh\fnc_init.sqf)
	[] L35 (x\cba\addons\xeh\fnc_init.sqf)
	[] L13 (z\ace\addons\interact_menu\XEH_preInit.sqf)
	[] L34 (/z/ace/addons/interact_menu/functions/fnc_compileMenu.sqf)
	[] L35 (/z/ace/addons/interact_menu/functions/fnc_compileMenu.sqf)

</details>

### CODE REVIEW

**Comment by dedmen** [comment]
Issue with variable assignment assembly has been fixed.
For some reason ASSIGNMENT and ASSIGNMENT_LOCAL have a different way the arguments are stored even though they are really the same thing...

https://github.com/dedmen/ArmaScriptCompiler/commit/19995559c0811db9a85c8c1f58a18db21ba6be3b

That should fix
 https://github.com/acemod/ACE3/blob/master/addons/common/functions/fnc_cbaSettings_loadFromConfig.sqf

**Comment by PabstMirror** [comment]
P:/z/ace/addons/common/functions/fnc_getMapGridData.sqfc doesn't compile because of
`private _zoomMax = 1e99;` which kinda makes sense as that's not valid in 32bit float

Could script compiler handle those as "positive infinity"?
or should we just swap those to 1e38 or something more sane
tbh, it's not a big deal if 2 files don't compile because it will always fall back to just use sqf

**Comment by kymckay** [comment]
@PabstMirror `1e39` should be positive infinity in SQF, whether compiler here knows about that idk

**Comment by kymckay** [comment]
Although it just occurred to me that might be what you meant, I think it should be possible to compile the infinity values since they must be informationally represented somehow when they appear in engine 🤔

**Comment by PabstMirror** [comment]
I think this is ready
could use review on changes to tools/make.py (bottom of files changed tab)
should continue to work on CI or systems without asc.exe

File: sqfc.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        "initsettings.sqf",
+        "initkeybinds.sqf",
+        "xeh_prep.sqf",
+        "\\dev\\",
+        "medical_ai\\statemachine.sqf",
+        "common\\functions\\fnc_dummy.sqf"
+    ],
+    "outputDir": "P:/",
+    "workerThreads": 12
+}
```
Comment (jonpas) [inline sqfc.json:18]:
```suggestion
}

```

File: sqfc.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+    "inputDirs": [
+        "P:/z/ace/addons/"
```
Comment (jonpas) [inline sqfc.json:3]: Can this not take relative directory?

**Comment by jonpas** [comment]
`make.py` changes look fine, although a bit confusing with the Exceptions instead of just `if`s, especially since you don't actually propagate the exception anywhere.

File: sqfc.json (hunk 2 above)
Comment (jokoho48) [reply sqfc.json:3]: https://github.com/dedmen/ArmaScriptCompiler/issues/8 saidly no, or atleast not yet

**Review by jonpas — APPROVED** [review]

**Pull request merged**